pkg/kp/dsstore/consul_store.go

```diff
@@ -326,6 +326,61 @@ func (s *consulStore) Watch(quitCh <-chan struct{}) <-chan WatchedDaemonSets {
 	return outCh
 }
 
+// WatchAll watches dsTree for all the daemon sets and returns a blocking
+// channel where the client can read a WatchedDaemonSetsList object which
+// contain all of the daemon sets currently on the tree
+func (s *consulStore) WatchAll(quitCh <-chan struct{}) <-chan WatchedDaemonSetList {
+	inCh := make(chan api.KVPairs)
+	outCh := make(chan WatchedDaemonSetList)
+	errCh := make(chan error, 1)
+
+	// Watch for changes in the dsTree and deletedDSTree
+	go consulutil.WatchPrefix(dsTree, s.kv, inCh, quitCh, errCh, time.Duration(250*time.Millisecond))
+
+	go func() {
+		defer close(outCh)
+		defer close(errCh)
+
+		var kvp api.KVPairs
+		for {
+			select {
+			case <-quitCh:
+				return
+			case err := <-errCh:
+				s.logger.WithError(err).Errorf("WatchPrefix returned error, recovered.")
```

Comment:
Hmm, OK. As long as no scheduling decisions are made on it. Because if the first thing received is an error, then this will send an empty list of daemon sets, right? I just didn't want someone thinking "that means there are no daemon sets. I will unschedule all of them", but I guess they'd use `Watch` for that.